### PR TITLE
Remove all traces of nodeagent

### PR DESCRIPTION
**Pull request by howardjohn**
Followup to https://github.com/istio/istio/pull/20777

### CODE REVIEW

**Review by ericvn — APPROVED** [review]

**Comment by rshriram** [comment]
What happens to the node agent on VMs?

**Comment by howardjohn** [comment]
@rshriram there are 2 node agents, one is node-agent-k8s for kubernetes and the other is node-agent for VMs. This only kills node-agent-k8s, the VM stuff is still there

**Review by ayj — APPROVED** [review]

**Review by rshriram — APPROVED** [review]

File: manifests/gateways/istio-ingress/templates/deployment.yaml [hunk 1]
```diff
@@ -212,13 +182,11 @@ spec:
             value: {{ .Values.global.jwtPolicy }}
           - name: PILOT_CERT_PROVIDER
             value: {{ .Values.global.pilotCertProvider }}
-{{- if or .Values.global.istiod.enabled $gateway.sds.enabled }}
+{{- if .Values.global.istiod.enabled }}
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply manifests/gateways/istio-ingress/templates/deployment.yaml]: I am going to remove the istiod flag next PR

File: manifests/gateways/istio-ingress/templates/deployment.yaml [hunk 2]
```diff
@@ -212,13 +182,11 @@ spec:
...
           - name: "ISTIO_META_USER_SDS"
             value: "true"
-{{- if .Values.global.istiod.enabled }}
           - name: CA_ADDR
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply manifests/gateways/istio-ingress/templates/deployment.yaml]: Good point

**Review by myidpt — APPROVED** [review]

**Pull request merged**